### DISCUSSION TITLE
Slider With Negative Value

### DISCUSSION THREAD

**Discussion by SDIjeremy** [alpinejs/alpine#3308] [7. Bugs (archived)]
The below code outputs 0 when ever the value is a negative number.

```html
<div x-data="{}" x-init="() => { console.log($refs.slider.value) }">
  <input type="range" name="slider" value="-100" x-ref="slider">
</div>
```
https://codepen.io/SDIjeremy/pen/BaVqRxK

**Comment by SimoTod** [comment]
https://www.w3schools.com/tags/att_input_type_range.asp

An input range goes from 0 to 100 unless yiu explicitly set min and max.

**Reply by SDIjeremy** [reply]
> https://www.w3schools.com/tags/att_input_type_range.asp
> 
> An input range goes from 0 to 100 unless yiu explicitly set min and max.

```html
<div x-data="{ min: -100, max: 100}" x-init="() => { console.log($refs.slider.value) }">
  <input type="range" name="slider" x-bind:min="min" x-bind:max="max" value="-100" x-ref="slider">
</div>
```
https://codepen.io/SDIjeremy/pen/BaVqRxK

It does in fact work when explicitly setting them, but when doing `x-bind` it does not.

**Reply by SimoTod** [reply]
Yep, it's the same as before. 

Value is set to -100 so it resolves to zero because min is 0 at that point. 

X-bind happens after the full page is ready. Changing a min attribute later doesn't force the value attribute again, html is not reactive. 

Either you use the standard attributes everywhere or you bind min max AND value. (but in that case the order matter, unless you have a real reason, you should just use min).

**Reply by ekwoka** [reply]
If you intend to have dynamic min/max, you could just set them to `-Infinity` and `Infinity` by default.
